src/main/java/io/cryostat/recordings/RecordingTargetHelper.java

```diff
@@ -132,13 +150,21 @@ public IRecordingDescriptor startRecording(
                     Optional<IRecordingDescriptor> previous =
                             getDescriptorByName(connection, recordingName);
                     if (previous.isPresent()) {
-                        if (!restart) {
+                        if (shouldRestartRecording(replace, previous.get())) {
+                            if (isRecordingStopped(previous.get())) {
+                                connection.getService().close(previous.get());
+                            } else {
+                                // If recording exists and running, stop and close it before
+                                // starting a
+                                // new one
+                                connection.getService().stop(previous.get());
```

Comment:
I think the recording can just be `close()`d even if it's still running, it probably isn't necessary to `stop()` it first. Should be easy to verify now that there's a nice robust test suite ;-)